lib/bolt/applicator.rb

```diff
@@ -206,7 +206,7 @@ def apply_ast(raw_ast, targets, options, plan_vars = {})
         end
 
         result_promises = targets.zip(futures).flat_map do |target, future|
-          @executor.queue_execute([target]) do |transport, batch|
+          @executor.queue_execute(target => nil) do |transport, batch|
```

Comment:
Could this... whole thing be replaced with a call to `run_task_with`? Hypothetically